### ISSUE TITLE
[Question] Multiple hard disk images at the same time using m3u?

### ISSUE THREAD

**Issue by vanfanel** [bbbradsmith/hatariB#37]
Hi there again, @bbbradsmith 

This time I come to you with what I expect to be a simple question: Is it possible to have at least two hard disk images loaded at the same time with an m3u file?

For example, if I create an m3u with:
```
disk1.gem
disk2.gem
```
...Only the second HD image is mounted.

Same if I mix GEM disks with SCSI images as in:
```
disk1.gem
disk2.scsi
```

Why would I need this? Well, because I am trying to have Spectre GCR (the Macintosh emulator) on a disk, and a Macintosh HD image as the second disk.

Is that possible at all?

**Comment by bbbradsmith** [comment]
There's no support for multiple hard drives currently.

Things you could do:
1. Place 2 partitions on one hard disk. This can't mix 2 different drive hardware formats, but at least provides 2 logical disks from the dame image.
2. Can Spectre be launched from a floppy instead?

**Comment by vanfanel** [comment]
Thanks @bbbradsmith , I was trying suggestion number two. Yes, Spectre can be launched from a floppy disk, it runs from an .st disk image with no problem at all.
Then I can create an .m3u file with these contents:
```
spectre.st
spectre.scsi
```
..where `spectre.st` is the floppy image containing the Spectre emulator and related files, and `spectre.scsi` is a Macintosh hard disk image with a System 5 installation, which should boot from Spectre.
This System 5 hard disk image is here: https://kdrive.infomaniak.com/app/share/156798/28ef00bd-f3ac-4556-8a5f-6353e6cfd6b2
And comes from this blog entry: https://obsolescenceguaranteed.blogspot.com/2013/04/atari-st-ultrasatan-spectre-mac.html

In other words, should HatariB be able to access an UltraSATAN SCSI hard disk image?

**Comment by bbbradsmith** [comment]
Okay, up to 4 hard drives should be supported as of: https://github.com/bbbradsmith/hatariB/commit/ed895c2b646685a2cbcc85ed6e59dec6250c44af

I have not yet tested this, as have test images prepared on this computer right now. If you're able to test it, please let me know the result.

Note that Hatari does not support multiple GemDOS directories, so your first example won't work. A GemDOS directory can, however, be divided into multiple partitions by creating folders within it `C/`, `D/`, `E/`, etc. If the partitions need to exist in different root directories, maybe you could accomplish that with symlinks?

**Comment by vanfanel** [comment]
@bbbradsmith Thanks for adding the multiple HDD functionality!
I have tried and indeed creating C, D, E... subdirs inside a GEM folders works, and they are presented as what seems to be different partitions.

However, creating an m3u with:
```
disc1.gem
disc2.scsi
```
...Doesn't seem to work, and Spectre isn't able to boot Mac System 5 from that scsi image.

Spectre is available here, in case you find this "emulator inside an emulator" experiment interesting:
[SPECTRE.EMU.zip](https://github.com/bbbradsmith/hatariB/files/14923379/SPECTRE.EMU.zip)

**Comment by bbbradsmith** [comment]
Does that setup work in Hatari stand-alone? (disc1 folder for GemDOS, and disc2.scsi image for the first SCSI hard drive)

**Comment by bbbradsmith** [comment]
Here's a test that adds two ACSI drives (partitioned in 1 and 2 parts: C, D, E) and two GemDOS partitions (G, H). It seems to work correctly.
[hdtest1.zip](https://github.com/bbbradsmith/hatariB/files/14938970/hdtest1.zip)

I'm not actually certain how to format a working SCSI image, so I haven't been able to test that. Anyway, if it works in Hatari it should also work here. Let me know if that isn't the case.

**Comment by bbbradsmith** [comment]
As far as I can tell, spectre expects an ACSI image, not an SCSI one. If you use the System 5 hard disk image you linked as an ACSI hard disk, and run SPECTRE.PRG, check off the auto mount hard disk option, and run it from file menu, it seems to get farther than if there's no hard disk. It will show a smiling mac icon for a bit, and then a crash report comes up. This is the same in Hatari stand-alone or hatariB, though.

**Comment by bbbradsmith** [comment]
Okay, well I'm going to close this because I think the hatariB question of supporting multiple hard drives has been resolved. I don't mind further comments if you manage to get Spectre running or if you have other information to share, but I feel like it shouldn't be an open issue at this point.

**Comment by vanfanel** [comment]
@bbbradsmith Yes, of course! Sorry, I should have closed this myself.
I will report any news regarding Spectre here. My next move is trying to contact the author of the blog, in case he can shed some light on this.

**Comment by vanfanel** [comment]
@bbbradsmith Got it to work! Chose 2MB of RAM in HATARI and it should boot. 
I'm using latest GIT standalone HATARI for that.
I was going to upload a disk image with a lot of nice Mac games, but compatibility of Spectre with games seems... non-existant.

**Comment by bbbradsmith** [comment]
Hmm, could you describe more of your setup for that? I have:
1. `100scsi image format.dsk` from the link, mounted as ACSI 0.
2. A floppy disk with the `SPECTRE.EMU` contents you linked.
3. Atari ST with TOS 1.04us, what TOS are you using? (Doesn't seem to change with various others.)
4. 2MB of memory.
5. Monochrome monitor.

I boot, open SPECTRE.PRG.
Check off "Automount HD"
Check off the partition "scsi100mb735"
Check the memory menu which shows 1792 K
Save settings
File > Spectre
Remove the floppy, click OK

It gets as far as briefly saying "welcome to machintosh" before this happens:
![screenshot](https://github.com/bbbradsmith/hatariB/assets/13027631/c5a888b9-75db-4934-8e69-a28fc2c1b605)

Are you doing something differently? Can you share your cfg?

**Comment by vanfanel** [comment]
@bbbradsmith Glad I could inspire your curiosity on this experiment :)
Spectre had a hardware component which was some kind of dongle that simply had the Macintosh roms on them. The version I pointed you to... well.. you know what I mean.

Have you tried any other games? Do they work for you? In my experience, compatibility seems to be veeeeeeeery low.

**Comment by bbbradsmith** [comment]
Hmm, maybe I had been lucky choosing Lode Runner. Have now tried: Dark Castle (black screen), Prince of Persia (reset after title screen appears), Shufflepuck Cafe (seemed to play fine, but without sound, until I tried to quit, then the CPU halted), Arkanoid (black screen).

**Comment by vanfanel** [comment]
Totally lucky, indeed. I get the same bad results here. Better stick to Atari ST I guess... :)

**Comment by bbbradsmith** [comment]
Hatari 2.5.0 is now merged, and it does seem to run spectre. :)

**Comment by vanfanel** [comment]
Congrats on merging 2.5.0! That's awesome, we have up-to-date ST emulation on RetroArch now, with RetroArch's trademark perfect video/audio synchronization! :)
Can't wait for this to be included on Lakka!
